File: src/models/task.js
const mongoose = require('mongoose')

const taskSchema = mongoose.Schema({
    description:{
        type:String,
        required:true
    },
    completed:{
        type:Boolean,
        default:false
    },
    completedAt:{
        type:Date
    },
    owner:{
        type:mongoose.Schema.Types.ObjectId,
        required:true,
        ref:'User'
    },
    category: {
    type: String
    },

    priority: {
        type: String
    },

    subtasks: [{
        type: String
    }],
},{
    timestamps: true
})

const Task = mongoose.model('Task', taskSchema)

module.exports=Task

File: backend-java/src/main/java/com/gaurav/taskmanager/model/Task.java
package com.gaurav.taskmanager.model;

import lombok.Data;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import java.time.Instant;
import java.util.List;

@Data
@Document(collection = "tasks")
public class Task {

    @Id
    private String id;

    private String description;
    private Boolean completed = false;
    private Instant completedAt;
    private Instant dueDate;
    private String category;
    private String priority;
    private List<String> subtasks;

    /** Owner user _id — matches Node backend field name */
    private String owner;

    @CreatedDate
    @Field("createdAt")
    private Instant createdAt;

    @LastModifiedDate
    @Field("updatedAt")
    private Instant updatedAt;
}
